lib/Model/AttendeeMapper.php

```diff
@@ -227,6 +221,8 @@ public function modifyPermissions(int $roomId, string $mode, int $newState): voi
 				Attendee::PERMISSIONS_LOBBY_IGNORE,
 			] as $permission) {
 				if ($permission & $newState) {
+					$query = $this->getModifyPermissionsBaseQuery($roomId);
```

Comment:
this could be outside the loop? We don't need to create the query builder 7 times?

Reply:
The query builder would be created as many times as individual permissions are added/removed, so it would be created seven times only if the seven permissions were added/removed at once.

A single query object needs to be created for each individual permission because [adding](https://github.com/nextcloud/spreed/blob/e8cff4fb7556a8811226e96bb5b7c7312f428e9a/lib/Model/AttendeeMapper.php#L259) or [removing](https://github.com/nextcloud/spreed/blob/e8cff4fb7556a8811226e96bb5b7c7312f428e9a/lib/Model/AttendeeMapper.php#L281) a single permission executes the statement. Is there any way to reset the query builder to reuse [the same base `update` query](https://github.com/nextcloud/spreed/blob/e8cff4fb7556a8811226e96bb5b7c7312f428e9a/lib/Model/AttendeeMapper.php#L206-L212) and change [the `set` and `andWhere` parts](https://github.com/nextcloud/spreed/blob/e8cff4fb7556a8811226e96bb5b7c7312f428e9a/lib/Model/AttendeeMapper.php#L241-L257)?